extracted/konan.serializer/src/org/jetbrains/kotlin/serialization/konan/SourceFileMap.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2010-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+package org.jetbrains.kotlin.serialization.konan
+
+import org.jetbrains.kotlin.descriptors.SourceFile
+import org.jetbrains.kotlin.serialization.deserialization.DeserializedPackageFragment
+
+private class DeserializedSourceFile(
+        val name_: String, val index: Int, val fragment: DeserializedPackageFragment) : SourceFile {
+    override fun getName(): String? = name_
+
+    override fun equals(other: Any?): Boolean {
+        return other is DeserializedSourceFile && fragment == other.fragment && index == other.index
```

Comment:
This is still incorrect if metadata contains multiple package fragments for one file.

Reply:
Not sure if this kind of correctness is critical for the practical use-cases. What may happen, is that two fragments of the same file are considered different, while they were in same original file. Will it break any important use-case?

Reply:
I am sure that this kind of correctness is critical for the practical use-cases.

> What may happen, is that two fragments of the same file are considered different, while they were in same original file. Will it break any important use-case?

It will break important use-case of using top-level declarations from this file. Different files must be translated into different Objective-C classes. As for now, one of these classes would have mangled name, but this behaviour should eventually be changed to reporting name clash during either compilation of Kotlin source code or compilation of Objective-C/Swift code that uses declarations from these files.

Reply:
OK, fixed.